### PR TITLE
Refactor search equipment

### PR DESCRIPTION
**Pull request by anistouri**

### CODE REVIEW

**Review by klesaulnier — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 1]
```diff
@@ -496,45 +485,39 @@ private List<EquipmentInfos> cleanRemovedEquipments(UUID networkUuid, String var
...
                 new TermsQuery.Builder().field(VARIANT_ID).terms(new TermsQueryField.Builder().value(List.of(FieldValue.of(initialVariantId), FieldValue.of(variantId))).build()).build();
 
+        BoolQuery.Builder boolQueryBuilder = new BoolQuery.Builder()
+                .filter(
+                        equipmentSearchQuery._toQuery(),
+                        networkUuidSearchQuery._toQuery(),
+                        variantIdSearchQuery._toQuery()
+                );
+
+        if (equipmentType != null && !equipmentType.isBlank()) {
```
Comment (thangqp) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:502]: Use !StringUtils.isEmpty() of Apache

Comment (anistouri) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Done

Comment (anistouri) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 2]
```diff
@@ -218,4 +222,93 @@ public void searchEquipmentInfosMultiVariants() {
         assertTrue(hits.contains(newGeneratorInfos));
         assertTrue(hits.contains(newLineInfos));
     }
+
+    @Test
+    public void searchModifiedEquipment() {
+
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:228]: Remove blank line ?

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 3]
```diff
@@ -218,4 +222,93 @@ public void searchEquipmentInfosMultiVariants() {
...
+        // Searching for the third version of the equipment and validating its presence in the results.
+        list = studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID_2, "loadId1", EquipmentInfosService.FieldSelector.ID, null, false);
+        assertEquals(1, list.size());
+        assertTrue(list.contains(loadInfos2));
+
+        // Re-searching for the second version of the equipment to ensure it is still retrievable without third version.
+        list = studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "loadId1", EquipmentInfosService.FieldSelector.ID, null, false);
+        assertEquals(1, list.size());
+        assertTrue(list.contains(loadInfos1));
+
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:259]: Remove blank line ?

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 4]
```diff
@@ -218,4 +222,93 @@ public void searchEquipmentInfosMultiVariants() {
...
+    }
+
+    @Test
+    public void testSearchForModifiedEquipmentsFilteredByType() {
+
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:264]: Remove blank line ?

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 5]
```diff
@@ -218,4 +222,93 @@ public void searchEquipmentInfosMultiVariants() {
...
+        // Re-verifying the retrieval of the first set of modified equipment for both LOAD and GENERATOR types without redandante.
+        list = studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "loadId1", EquipmentInfosService.FieldSelector.ID, "LOAD", false);
+        assertEquals(1, list.size());
+        assertTrue(list.contains(loadInfos1));
+        list = studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_g1", EquipmentInfosService.FieldSelector.ID, "GENERATOR", false);
+        assertEquals(1, list.size());
+        assertTrue(list.contains(generatorInfos1));
+    }
+
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:313]: Remove blank line ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java [hunk 6]
```diff
@@ -52,9 +55,17 @@
...
     private static final String TEST_FILE = "testCase.xiidm";
+    private static final String EQUIPMENT_TYPE = "equipmentType.keyword";
+    private static final String EQUIPMENT_ID = "equipmentId.fullascii";
+    private static final String EQUIPMENT_ID_RAW_FIELD = "equipmentId.raw";
+    private static final String EQUIPMENT_NAME_RAW_FIELD = "equipmentName.raw";
 
-    private static final UUID NETWORK_UUID = UUID.randomUUID();
-    private static final UUID NETWORK_UUID_2 = UUID.randomUUID();
+    private static final String EQUIPMENT_NAME_FULLASCII_FIELD = "equipmentName.fullascii";
+    private static final String EQUIPMENT_NAME = "equipmentName";
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java]: Done

File: src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java [hunk 7]
```diff
@@ -52,9 +55,17 @@
 public class EquipmentInfosServiceTests {
 
     private static final String TEST_FILE = "testCase.xiidm";
+    private static final String EQUIPMENT_TYPE = "equipmentType.keyword";
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java]: Done

File: src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java [hunk 8]
```diff
@@ -52,9 +55,17 @@
...
+    private static final String EQUIPMENT_ID = "equipmentId.fullascii";
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java (hunk 2 above)
Comment (anistouri) [reply src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java (hunk 4 above)
Comment (anistouri) [reply src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java]: Done

**Pull request merged**